pkg/destroy/aws/aws.go

```diff
@@ -504,6 +525,65 @@ func deleteEC2Image(client *ec2.EC2, id string, logger logrus.FieldLogger) error
 		return err
 	}
 
+	logger.Info("Deleted")
+
+	// need to delete the snapshots after deleting the ami. aws will block the
+	// snapshot deletion with an InvalidSnapshot.InUse error until the ami is deleted.
+	return deleteEC2Snapshots(client, snapshots, logger)
```

Comment:
Instead of looping over snapshots inside `deleteEC2Image`, how about we actively tag the AMI's snapshots before deleting the AMI?  Then we can deregister the AMI and exit `deleteEC2Image`.  The snapshots will get picked up by future loops of the tag-lister.  That also protects us from leaking if somebody kills the teardown code after the AMI is removed but before the snapshots are gone, because a new teardown attempt with the same parameters would still pick up the snapshots by tag.

Reply:
> Instead of looping over snapshots inside deleteEC2Image, how about we actively tag the AMI's snapshots before deleting the AMI?

I've pushed two fixup commits with this approach, although I haven't had time to test them yet.  Pull with:

```console
$ git pull git://github.com/wking/openshift-installer snapshot-fixups
```

and squash 81d967c6c if it looks good (there's also a drive-by `Wrapf` -> `Wrap` fixup, which you can take or leave).  The diff is fairly noisy from plumbing `filter` through.  I plumbed it through the whole chain for consistency, but if you want a smaller diff you could just plumb it through `deleteARN` -> `deleteEC2` -> `deleteEC2Image` (which you can get from my branch by squashing in eadbf6bc9 as well).

Reply:
Done in c07b1ed to fcade11.